### PR TITLE
feat(mesh): access token

### PR DESCRIPTION
**Pull request by tyc67**
## Notable Changes

- integrate with `/accesstoken` api [doc](https://paper.dropbox.com/doc/Mesh--CVkR5zNs546EgrTJEW3OqGcwAg-csgs0E6jyRfz2i8JYivX4)
- Updated `handleFirebaseToken()` to validate Firebase tokens and authenticate via `getAccessToken()`
- Enhanced `handleSignIn()` to include Firebase persistence state checks for across sessions

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208015815768755

### CODE REVIEW

File: packages/mesh/app/actions/auth.ts [hunk 1]
```diff
@@ -54,14 +54,50 @@ export async function clearTokenCookie() {
...
+  if (decodedAccessToken?.exp) {
+    tokenAge = Math.floor(decodedAccessToken.exp - Date.now() / 1000)
+  }
+
+  cookies().set('token', accessToken, {
+    path: '/',
+    httpOnly: true,
+    sameSite: 'lax',
+    secure: true,
+    maxAge: tokenAge,
```
Comment (erase2004) [inline packages/mesh/app/actions/auth.ts]: 可以改用 expires 處理

**Review by erase2004 — APPROVED** [review]
有注意到登入後，畫面呈現為沒有使用者的狀態，這部份之後需要再調整
Snipaste_2024-08-26_20-29-33
Snipaste_2024-08-26_20-30-10

**Review by LinHeMa — APPROVED** [review]

File: packages/mesh/app/actions/auth.ts (hunk 1 above)
Comment (tyc67) [reply packages/mesh/app/actions/auth.ts]: 修改於 [a047805](https://github.com/readr-media/Sachiel/pull/833/commits/a0478055d740947ebcd22f353725f7cead4cf06a)

**Comment by tyc67** [comment]
> 有注意到登入後，畫面呈現為沒有使用者的狀態，這部份之後需要再調整 Snipaste_2024-08-26_20-29-33 Snipaste_2024-08-26_20-30-10

登入後更新 context user state [81b0fef](https://github.com/readr-media/Sachiel/pull/833/commits/81b0feff8b5bb0d2b7e1c93f6c446c67fc9c37dc)

**Pull request merged**